--- submodules/cryptattacktester-20231020/uniformmatrix_orig.cpp	2026-02-07 15:07:44.151132173 +0100
+++ submodules/cryptattacktester-20231020/uniformmatrix.cpp	2026-02-07 15:06:39.642823169 +0100
@@ -24,16 +24,19 @@
     bigint q = 2;
     while (q < N) { ++lgq; q *= 2; }
 
-    bigint Wmin = 0;
-    bigint Wmax = N;
-    selection_constrain(S,"W",Wmin,Wmax);
-    for (bigint W = Wmin;W <= Wmax;++W) {
-      bigint K = N-lgq*W;
-      if (10*K < 7*N) break;
-      if (10*K > 8*N) continue;
-      if (!selection_allows(S,"K",K.get_str())) continue;
+    bigint Kmin = 4;
+    bigint Kmax = 4096;
+    selection_constrain(S, "K", Kmin, Kmax);
+    for (bigint K = Kmin; K <= Kmax; ++K) {
+      // bigint K = N/2;
+      // if (!selection_allows(S,"K",K.get_str())) continue;
 
-      result.push_back(vector<bigint> {N,K,W});
+      bigint Wmin = 0;
+      bigint Wmax = N;
+      selection_constrain(S,"W",Wmin, Wmax);
+      for (bigint W = Wmin;W<=Wmax;++W) {
+        result.push_back(vector<bigint>{N, K, W});
+      }
     }
   }
